test/integration/src/py/integration_test_context.py

```diff
@@ -0,0 +1,211 @@
+
+
+class IntegrationTestContext:
+    def __init__(self, snapshot_name):
+        self.cmd = make_py_module.Integrator()
+        self.it_playbook = make_py_module.IntegrationTestsPlaybook(self.cmd)
+        self.processes = self.it_playbook.restore_snapshot(snapshot_name)
+
+    def get_required_var(self, varname):
+        return test_utilities.get_required_env_var(varname)
+
+    def get_optional_var(self, varname, default_value):
+        return test_utilities.get_optional_env_var(varname, default_value)
+
+    def env_or_truffle_artifact(self, contract_name, contract_env_var, smart_contract_artifact_dir, ethereum_network_id):
+        result = self.get_optional_var(contract_env_var, None)
+        return result if result else test_utilities.contract_address(
+            smart_contract_artifact_dir=smart_contract_artifact_dir,
+            contract_name=contract_name,
+            ethereum_network_id=ethereum_network_id
+        )
+
+    @property
+    def sifnoded_node(self):
+        return self.get_optional_var("SIFNODE", None)
+
+    @property
+    def sifnode_base_dir(self):
+        return self.get_required_var("BASEDIR")
+
+    @property
+    def smart_contracts_dir(self):
+        return self.get_optional_var("SMART_CONTRACTS_DIR", os.path.join(self.sifnode_base_dir, "smart-contracts"))
+
+    @property
+    def smart_contract_artifact_dir(self):
+        result = self.get_optional_var("SMART_CONTRACT_ARTIFACT_DIR", None)
+        return result if result else os.path.join(self.smart_contracts_dir, "build/contracts")
+
+    @property
+    def bridgebank_address(self):
+        return self.env_or_truffle_artifact("BridgeBank", "BRIDGE_BANK_ADDRESS", self.smart_contract_artifact_dir,
+            self.ethereum_network_id)
+
+    @property
+    def is_ropsten_testnet(self):
+        """if sifnode_clinode is set, we're talking to ropsten/sandpit"""
+        return bool(self.sifnoded_node)
+
+    @property
+    def ethereum_network_id(self):
+        result = self.get_optional_var("ETHEREUM_NETWORK_ID", None)
+        if result:
+            return result
+        else:
+            return 3 if self.is_ropsten_testnet else 5777
+
+    @property
+    def bridgetoken_address(self):
+        return self.env_or_truffle_artifact("BridgeToken", "BRIDGE_TOKEN_ADDRESS", self.smart_contract_artifact_dir,
+            self.ethereum_network_id)
+
+    @property
+    def ethereum_network(self):
+        return self.get_optional_var("ETHEREUM_NETWORK", "")
+
+    @property
+    def chain_id(self):
+        return self.get_optional_var("DEPLOYMENT_NAME", "localnet")
+
+    @property
+    def is_ganache(self):
+        """true if we're using ganache"""
+        return not self.ethereum_network
+
+    @property
+    def sifchain_fees_int(self):
+        return 200000
+
+    @property
+    def sifchain_fees(self):
+        """returns a string suitable for passing to sifnoded"""
+        return f"{self.sifchain_fees_int}rowan"
```

Comment:
This code was copied verbatim from existing integration tests. The above values are used for `--fees` parameter, and `--gas auto` is always added as well.

@timlind I can remove the `--fees` parameter and add `--gas-prices=0.5rowan --gas-adjustment=1.5` instead. Is this what you mean? Can we assume that, if we make this change and it doesn't break any tests, it would be a preferred solution?

Reply:
Note: it doesn't work, i.e. it breaks several tests.

Reply:
I'm not sure if I understood Tim's point correctly, but it seems to me that this change is not trivial and it would best be done together with refactoring integration tests. Weighing between two options I'd say that it's better to have integration tests working for now, especially since this implies that transactions are not running out of gas. But I will tag this function as obsolete so that we'll remember to replace it.